proposals/lip-0051.md

```diff
@@ -1267,46 +1175,94 @@ unlock(address, moduleID, tokenID, amount):
         or lockedAmount(address, moduleID, tokenID) < amount):
         unlock fails
 
-    availableBalance(address, tokenID) += amount
     lockedAmount(address, moduleID, tokenID) -= amount
+    availableBalance(address, tokenID) += amount
+```
+
+#### forwardMessageFee
+
+This function is only called during the execution of a cross-chain update on the mainchain.
+```python
+forwardMessageFee(ccm):
+    fee = ccm.fee
+    if (fee < 0
+        or getChannel(ccm.sendingChainID).messageFeeTokenID != getChannel(ccm.receivingChainID).messageFeeTokenID):
```

Comment:
As far as I understand `getChannel` is an API of the interoperability module so we should use the notation `Interoperability.getChannel` (same also in other places).

Reply:
Indeed, added.